### DISCUSSION TITLE
Is getInitialProps useful for this case?

### DISCUSSION THREAD

**Discussion by fromancient** [vercel/next.js#19195] [Help]
I have been working on a project already developed by the other dev. But I am wondering about how he used getInitialProps in the project. 
Here is the some code part to use getInitialProps for prefetching in the page. This is really not a prefetch. 
`
Index.getInitialProps = async ({ store }: PageContextProps ) => {
  try {
    await store.dispatch(
      fetchPosts(settings.blogs, { per_page: settings.postsPerPage })
    );
    return {};
  } catch (e) {
    console.error(e);
    return {};
  }
};
`

I want to hear your experts' overview about this code part and also about the best usecase of getInitialProps.
Regards

**Comment by ljosberinn** [comment]
getInitialProps to me has no use case whatsoever anymore. getServerSideprops/getStaticProps in combination with reusable functions extending them should be used over gIP. The docs recommend so too.

**Comment by fromancient** [comment]
Thanks for your kind reply
But I am just wondering how I can use redux store in getServerSideProps
Do you have any idea?

**Reply by ljosberinn** [reply]
https://github.com/vercel/next.js/blob/canary/examples/with-redux/pages/ssr.js
